src/components/NetworkModal/index.tsx

```diff
@@ -191,6 +191,13 @@ export default function NetworkModal(): JSX.Element | null {
     }, [error, network])
 
     const isMetaMask = window.ethereum && window.ethereum.isMetaMask
+    const isOnlyMeta = window.ethereum?.providers?.length === 1
+    if (!isOnlyMeta) {
+      const provider = window.ethereum?.providers.find((isMetaMask: any) => isMetaMask.isMetaMask)
+      if (window.ethereum){
+        window.ethereum.selectedProvider = provider
+      }
+    }
```

Comment:
lets simplify this
```suggestion
      if(window.ethereum)
      {
         const provider = window.ethereum.providers.find((isMetaMask: any) => isMetaMask.isMetaMask) || window.ethereum.providers[0]
      const isMetaMask = provider?.isMetaMask
      window.ethereum.selectedProvider = provider
    }
```

Reply:
these suggested changes, imo, are to zoomed in
cause when there are no multiple providers, there actually doesn't exist a ethereum.providers
then metamask just exist top level in ethereum object
window.ethereum.providers[0] --> so this would return undefined

The more general question is and solutions are:
Exclusively MetaMask: We should define a more global ethereum object, where Meta is always set as the selected provider
Support for both: Then the logic has to change in a few places to actually support the different scopes

it could also just be that the interface should be defined slightly different
I also believe the scopes could be defined in a more global way, to get rid of repetitiveness 

but the first question is, which do we want to support?

Reply:
then simply change `|| window.ethereum.providers[0]` to `|| window.ethereum.providers?[0] || window.ethereum`
i want to keep it simple, if metamask exists among multiple providers it will use metamask otherwise it will use the first provider available that's it.
the wallet_X are for all I know metamask only.